src/helics/application_api/ValueFederateManager.cpp

```diff
@@ -125,9 +125,21 @@ void ValueFederateManager::addTarget(const Publication& pub, const std::string&
 
 void ValueFederateManager::addTarget(const Input& inp, const std::string& target)
 {
+    {
+        auto iTHandle = inputTargets.lock();
+        auto rng = iTHandle->equal_range(inp.handle);
+        for (auto el = rng.first; el != rng.second; ++el) {
+            if (el->second == target) {
+                fed->logWarningMessage(std::string("Duplicate input targets detected for ") +
+                                       inp.actualName + "::" + target);
+                return;
+            }
+        }
+    }
+
     coreObject->addSourceTarget(inp.handle, target);
+    auto it = inputTargets.lock()->emplace(inp.handle, target);
```

Comment:
Are there plans to use the `iterator it` returned by emplace at a later date? Not technically a problem, but the storage of an immediately deleted temporary seemed weird.